src/Cli/dotnet/commands/RestoringCommand.cs

```diff
@@ -76,19 +76,61 @@ private static IEnumerable<string> Prepend(string argument, IEnumerable<string>
         private static bool HasArgumentToExcludeFromRestore(IEnumerable<string> arguments)
             => arguments.Any(a => IsExcludedFromRestore(a));
 
-        private static readonly string[] propertyPrefixes = new string[]{ "-", "/", "--" };
+        private static readonly string[] propertyPrefixes = new string[] { "-", "/", "--" };
 
-        private static bool IsExcludedFromRestore(string argument)
-            => propertyPrefixes.Any(prefix => argument.StartsWith($"{prefix}property:TargetFramework=", StringComparison.Ordinal)) ||
-               propertyPrefixes.Any(prefix => argument.StartsWith($"{prefix}p:TargetFramework=", StringComparison.Ordinal));
+        // these properties trigger a separate restore
+        private static List<string> PropertiesToExcludeFromRestore =
+        [
+            "TargetFramework"
+        ];
 
         //  These arguments don't by themselves require that restore be run in a separate process,
         //  but if there is a separate restore process they shouldn't be passed to it
+        private static List<string> FlagsToExcludeFromRestore =
+        [
+            "getProperty",
+            "getItem",
+            "getTargetResult",
+            "t",
+            "target",
+            "consoleloggerparameters",
+            "clp"
+        ];
+
+        private static List<string> FlagsToExcludeFromSeparateRestore =
+            ComputeFlags(FlagsToExcludeFromRestore).ToList();
+        private static List<string> PropertiesToExcludeFromSeparateRestore =
+            ComputePropertySwitches(PropertiesToExcludeFromRestore).ToList();
+
+        private static IEnumerable<string> ComputePropertySwitches(List<string> properties)
+        {
+            foreach (var prefix in propertyPrefixes)
+            {
+                foreach (var property in properties)
+                {
+                    yield return $"{prefix}property:{property}=";
+                    yield return $"{prefix}p:{property}=";
+                }
+            }
+        }
+
+        private static IEnumerable<string> ComputeFlags(List<string> flags)
+        {
+            foreach (var prefix in propertyPrefixes)
```

Comment:
Can we change the name of 'propertyPrefixes'? I got stuck for a minute in thinking this was like -p:getProperty:foo before I realized it was really just a generic switch prefix.

Reply:
Can do! And great note about the restore loggers as well - we should pass on a minimal verbosity and stuff if they are passed in.

Reply:
Sure - I was going for relatively minimal diffs since this would be a servicing fix, but that's a small rename.